packages/core/src/transports/multiplexed.ts

```diff
@@ -0,0 +1,84 @@
+import type {
+  BaseTransportOptions,
+  Envelope,
+  Event,
+  EventItem,
+  Transport,
+  TransportMakeRequestResponse,
+} from '@sentry/types';
+import { dsnFromString, forEachEnvelopeItem } from '@sentry/utils';
+
+import { getEnvelopeEndpointWithUrlEncodedAuth } from '../api';
+
+interface MatchParam {
+  // The envelope to be sent
+  envelope: Envelope;
+  // A function that returns an event from the envelope if one exists
+  getEvent(): Event | undefined;
+}
+
+type Matcher = (param: MatchParam) => string[];
+
+function eventFromEnvelope(env: Envelope): Event | undefined {
+  let event: Event | undefined;
+
+  forEachEnvelopeItem(env, (item, type) => {
+    if (type === 'event') {
```

Comment:
I am not 100% sure, but is the intent here to only match error events? Or should we also match other event types? 

IMHO at least all sub types of `Event` should probably be matched, so event, transaction, profile, replay_recording?

Reply:
After chatting with @AbhiPrasad offline, let's filter out any `Event` subtype here:
* event
* transaction
* profile
* replay_event

So we get an `Event` out here for sure. 

And then, we can add an optional third argument to the `makeMultiplexedTransport` that allows to filter events that should be matched:

```tsfunction makeMultiplexedTransport<TO extends BaseTransportOptions>(
  createTransport: (options: TO) => Transport,
  matcher: Matcher,
  eventFilter?: (event: Event) => boolean
): (options: TO) => Transport
```

When not provided, we default to

```js
eventFilter: (event: Event) => !event.type
```

Which will only multiplex errors by default.

Does that sound reasonable?

Reply:
Adding `eventFilter` means we'd be passing passing two different callbacks to `makeMultiplexedTransport ` which seems far too complex when the same can be achieved via a single callback. 

The matcher callback already gets passed an object containing the `envelope` and a `getEvent` shortcut function. 

We could modify the `getEvent` function to allow users to pass the types they're interested in:
```ts
interface MatchParam {
  envelope: Envelope;
  getEvent(...types: string[] = ['event']): Event | undefined
}
```

Then you could create a matcher function to capture all event types to a specific DSN like this:
```ts
const myMatcher =  ({ getEvent }) => getEvent('event', 'transaction', 'profile', 'replay_event') ? [DSN1] : []
```

Reply:
Modifying `getEvent` makes sense to me!